libzerocash/Address.cpp

```diff
@@ -11,24 +11,35 @@
  * @copyright  MIT license (see LICENSE file)
  *****************************************************************************/
 
-#include <cryptopp/osrng.h>
-using CryptoPP::AutoSeededRandomPool;
+#include "Zerocash.h"
+#include "Address.h"
 
-#include <cryptopp/eccrypto.h>
-using CryptoPP::ECP;
-using CryptoPP::ECIES;
+namespace libzerocash {
 
-#include <cryptopp/oids.h>
-namespace ASN1 = CryptoPP::ASN1;
+PrivateAddress::PrivateAddress() {
+    this->sk_enc = "";
+}
 
-#include <cryptopp/filters.h>
-using CryptoPP::StringSink;
-using CryptoPP::StringStore;
+void PrivateAddress::createPrivateAddress(const std::vector<unsigned char> a_sk, const std::string sk_enc) {
+    this->a_sk = a_sk;
+    this->sk_enc = sk_enc;
+}
 
-#include "Zerocash.h"
-#include "Address.h"
+bool PrivateAddress::operator==(const PrivateAddress& rhs) const {
+	return ((this->a_sk == rhs.a_sk) && (this->sk_enc == rhs.sk_enc));
+}
 
-namespace libzerocash {
+bool PrivateAddress::operator!=(const PrivateAddress& rhs) const {
+	return !(*this == rhs);
+}
+
+const std::string PrivateAddress::getEncryptionSecretKey() const {
+    return this->sk_enc;
+}
+
+const std::vector<unsigned char>& PrivateAddress::getAddressSecret() const {
```

Comment:
Since it's called "addressSecret" upstream, I don't think we should change it to avoid having to maintain a larger-than-necessary delta.